presto-main/src/test/java/com/facebook/presto/sql/planner/iterative/rule/test/RuleAssert.java

```diff
@@ -240,5 +296,10 @@ public PlanNode getResult()
         {
             return result.getTransformedPlan().orElseThrow(() -> new IllegalStateException("Rule was not applied"));
         }
+
+        public <T extends Trait> boolean satisfies(T trait)
+        {
+            return lookup.isTraitSatisfied(planNode, trait);
```

Comment:
This doesn't mean `RuleApplication` satisfies anything. The trait could be satisfied before rule was applied. You should instead:
```
this.result.getTrait().get().satisfies(trait)
```